nexmark-flink/src/main/java/com/github/nexmark/flink/metric/MetricReporter.java

```diff
@@ -154,29 +168,30 @@ public JobBenchmarkMetric reportMetric(String jobId, long eventsNum) {
 		}
 		double sumTps = 0.0;
 		double sumCpu = 0.0;
+		int realMetricSize = metrics.size();
 
 		for (BenchmarkMetric metric : metrics) {
+			if (Double.compare(metric.getTps(), 0.0) == 0) {
```

Comment:
Only backmost zero should be removed.

Reply:
I don't think this has much impact on the TPS calculation, we can ignore this optimisation for now.